pkg/controller/atlascluster/cluster.go

```diff
@@ -38,17 +44,66 @@ func ensureClusterState(wctx *workflow.Context, connection atlas.Connection, pro
 
 	switch c.StateName {
 	case "IDLE":
-		return c, workflow.OK()
+		spec, err := cluster.Spec.Cluster()
+		if err != nil {
+			return c, workflow.Terminate(workflow.Internal, err.Error())
+		}
+
+		if done, err := clusterMatchesSpec(wctx, c, cluster.Spec); err != nil {
+			return c, workflow.Terminate(workflow.Internal, err.Error())
+		} else if done {
+			return c, workflow.OK()
+		}
+
+		c, _, err = client.Clusters.Update(ctx, project.Status.ID, cluster.Spec.Name, spec)
+		if err != nil {
+			return c, workflow.Terminate(workflow.ClusterNotCreatedInAtlas, err.Error())
+		}
+
+		return c, workflow.InProgress(workflow.ClusterNotUpToDate, "cluster update started")
 
 	case "CREATING":
-		return c, workflow.InProgress(workflow.ClusterCreating, "Cluster is provisioning")
+		return c, workflow.InProgress(workflow.ClusterCreating, "cluster is provisioning")
```

Comment:
[nit] why lowercase? I believe all status fields are in common start from Uppercase

Reply:
most of our status fields are `err.Error()`, which, by convention, is lowercase

Reply:
ah I see. Then it makes sense to uppercase them before writing to the status.
Ok then 👍

Reply:
we could also prepend something like `"Error: " + err.Error()` to simplify things